packages/puppeteer-core/src/cdp/Accessibility.ts

```diff
@@ -508,9 +509,30 @@ class AXNode {
         if (!this.payload.backendDOMNodeId) {
           return null;
         }
-        return (await this.#realm.adoptBackendNode(
+        const handle = await this.#realm.adoptBackendNode(
           this.payload.backendDOMNodeId,
-        )) as ElementHandle<Element>;
+        );
+
+        if (!handle || !(handle instanceof ElementHandle)) {
+          return null;
+        }
+
+        const isTextNode = await handle.evaluate(node => {
+          return node.nodeType === Node.TEXT_NODE;
+        });
+        // Text nodes are not elements, so we can't return a handle to them.
+        // Instead, we find the parent element and return a handle to the parent.
+        if (isTextNode) {
+          using parent = await handle.evaluateHandle(node => {
+            return node.parentElement;
+          });
+
+          void handle.dispose().catch(debugError);
+
+          return parent.move() as ElementHandle<Element>;
+        }
+
+        return handle;
```

Comment:
```suggestion
        return  handle.evaluateHandle(node =>{
          if (node.nodeType === Node.TEXT_NODE) {
            return node.parentElement;
          }
          return node;
        });
```

let's combine it into one call?

Reply:
yes, I like this